src/App.tsx

```diff
@@ -1,61 +1,152 @@
+import { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer, { User } from './api/users';
+import todosFromServer from './api/todos';
+
+type Todo = {
+  id: number,
+  title: string,
+  completed: boolean,
+  userId: number,
+  user: User,
+};
+
+type Props = {
+  users: (Todo[] | undefined) | undefined;
+};
+
+const getDefaultUser = (): User => ({
+  id: -1,
+  name: 'Unknown User',
+  username: '',
+  email: '',
+});
+
+const getUser = (id: number) => {
+  const foundUser = usersFromServer.find((user) => (
+    user.id === id));
+
+  return foundUser || getDefaultUser();
+};
+
+const getUserFromName = (name: string) => {
+  const foundUser = usersFromServer.find((user) => (
+    user.name === name));
+
+  return foundUser;
+};
+
+const visibleTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.id),
+}));
+
+export const App: React.FC<Props> = () => {
+  const [name, setName] = useState('Choose a user');
+  const [title, setTitle] = useState('');
+  const [nameError, setNameError] = useState(false);
+  const [titleError, setTitleError] = useState(false);
+  const [todos, setTodos] = useState(visibleTodos);
+
+  const handleSelection = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setName(event.target.value);
+  };
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (name === 'Choose a user') {
+      setNameError(true);
+    }
+
+    if (!title) {
+      setTitleError(true);
+    }
+
+    const newUser = getUserFromName(name);
+
+    if (!newUser) {
+      setNameError(true);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: todos.length + 1,
+      title,
+      completed: false,
+      userId: newUser.id,
+      user: newUser || undefined,
+    };
+
+    setTodos([...visibleTodos, newTodo]);
+    setName('');
+    setTitle('');
+
+    visibleTodos.push(newTodo);
```

Comment:
Modifying 'visibleTodos' directly is not recommended. Use setTodos to update todos state.